llvm/lib/Target/AMDGPU/SIShrinkInstructions.cpp

```diff
@@ -159,7 +159,7 @@ bool SIShrinkInstructions::shouldShrinkTrue16(MachineInstr &MI) const {
 }
 
 bool SIShrinkInstructions::isKImmOperand(const MachineOperand &Src) const {
-  return isInt<16>(Src.getImm()) &&
+  return isInt<16>(SignExtend64(Src.getImm(), 32)) &&
```

Comment:
Could write this as `isInt<16>((int32_t)Src.getImm())` but it's a matter of taste I guess.

Reply:
> Could write this as `isInt<16>((int32_t)Src.getImm())` but it's a matter of taste I guess.

Same thing in principle.